File: .pi/skills/data-viz/scripts/lab_style.R
#!/usr/bin/env Rscript
# CHATLabAI lab style for ggplot2 figures.
# Mirrors scripts/lab_style.py: palette, fonts, sizing, theme.
#
# Usage:
#   source("scripts/lab_style.R")
#   ggplot(data, aes(x, y)) + geom_point() + theme_lab()
#
# palette_lab() returns the lab color vector; theme_lab() returns a ggplot2 theme.

# ---- Lab palette (Okabe-Ito inspired, colorblind-safe) ----------------------
palette_lab <- function() {
  c("#0072B2", "#D55E00", "#009E73", "#CC79A7",
    "#F0E442", "#56B4E9", "#E69F00", "#999999")
}

# ---- Lab theme --------------------------------------------------------------
# A clean, publication-ready theme matching the Python matplotlib style.
theme_lab <- function(base_size = 11, base_family = "sans") {
  if (!requireNamespace("ggplot2", quietly = TRUE)) {
    stop("[data-viz] ggplot2 is required. Install with: install.packages('ggplot2')")
  }
  ggplot2::theme_minimal(base_size = base_size, base_family = base_family) +
    ggplot2::theme(
      plot.title = ggplot2::element_text(size = base_size + 1, face = "bold", hjust = 0),
      axis.title = ggplot2::element_text(size = base_size),
      axis.text  = ggplot2::element_text(size = base_size - 1, color = "#333333"),
      axis.line  = ggplot2::element_line(color = "#333333", linewidth = 0.6),
      panel.grid.major = ggplot2::element_line(color = "#DDDDDD", linewidth = 0.4),
      panel.grid.minor = ggplot2::element_blank(),
      panel.background = ggplot2::element_rect(fill = "white", color = NA),
      plot.background  = ggplot2::element_rect(fill = "white", color = NA),
      legend.position  = "bottom",
      legend.title     = ggplot2::element_blank(),
      legend.background = ggplot2::element_blank()
    )
}

# ---- Default export helper --------------------------------------------------
# Save a ggplot to PNG + SVG + PDF at publication DPI.
save_lab <- function(plot, basename, width = 6, height = 4, dpi = 300) {
  if (!requireNamespace("ggplot2", quietly = TRUE)) {
    stop("[data-viz] ggplot2 is required.")
  }
  for (ext in c("png", "svg", "pdf")) {
    path <- paste0(basename, ".", ext)
    ggplot2::ggsave(path, plot = plot, width = width, height = height, dpi = dpi,
                    device = if (ext == "svg") ggplot2::svglite::svglite else NULL)
  }
}

# ---- Demo (run this script standalone to sanity-check the style) -----------
if (sys.nframe() == 0L) {
  if (!requireNamespace("ggplot2", quietly = TRUE)) {
    cat("[data-viz] ggplot2 not installed; install with: install.packages('ggplot2')\n")
    quit(status = 0)
  }
  library(ggplot2)
  df <- data.frame(
    group = rep(c("A", "B", "C"), each = 10),
    score = c(rnorm(10, 5, 1), rnorm(10, 7, 1), rnorm(10, 6, 1.2))
  )
  p <- ggplot(df, aes(group, score, fill = group)) +
    geom_violin() +
    geom_jitter(width = 0.1, color = "#333333", alpha = 0.5) +
    scale_fill_manual(values = palette_lab()) +
    labs(title = "CHATLabAI lab style (R)") +
    theme_lab()
  cat("[data-viz] lab_style.R loaded. theme_lab() and palette_lab() available.\n")
  invisible(p)
}

File: .pi/skills/data-viz/scripts/lab_style.py
#!/usr/bin/env python3
"""CHATLabAI lab style for matplotlib figures.

Defines a consistent visual style (palette, fonts, sizing, DPI) so every figure produced by
the data-viz skill matches. Importable by plot.py and brainmap.py:

    from lab_style import apply_lab_style, LAB_STYLE
    apply_lab_style()
"""
from __future__ import annotations

import matplotlib
matplotlib.use("Agg")  # headless; must be set before pyplot import
import matplotlib as mpl
import matplotlib.font_manager as fm

__all__ = ["LAB_STYLE", "apply_lab_style", "LAB_PALETTE"]

# Colorblind-safe palette (Okabe-Ito inspired) tuned for a neuroaesthetics lab.
LAB_PALETTE = [
    "#0072B2",  # blue
    "#D55E00",  # vermillion
    "#009E73",  # bluish green
    "#CC79A7",  # reddish purple
    "#F0E442",  # yellow
    "#56B4E9",  # sky blue
    "#E69F00",  # orange
    "#999999",  # grey
]

LAB_STYLE = {
    "figure.figsize": (6.0, 4.0),
    "figure.dpi": 150,
    "savefig.dpi": 300,
    "savefig.bbox": "tight",
    "font.family": "sans-serif",
    "font.sans-serif": ["Helvetica", "Arial", "DejaVu Sans"],
    "font.size": 11,
    "axes.titlesize": 12,
    "axes.labelsize": 11,
    "axes.edgecolor": "#333333",
    "axes.linewidth": 0.8,
    "axes.grid": True,
    "grid.color": "#DDDDDD",
    "grid.linewidth": 0.5,
    "axes.spines.top": False,
    "axes.spines.right": False,
    "xtick.labelsize": 10,
    "ytick.labelsize": 10,
    "legend.frameon": False,
    "legend.fontsize": 10,
}


def apply_lab_style() -> dict:
    """Apply the CHATLabAI matplotlib style and return the style dict.

    Safe to call multiple times. Also registers a color cycle from LAB_PALETTE.
    """
    mpl.rcParams.update(LAB_STYLE)
    mpl.rcParams["axes.prop_cycle"] = mpl.cycler(color=LAB_PALETTE)
    return LAB_STYLE


if __name__ == "__main__":
    apply_lab_style()
    print("CHATLabAI lab style applied.")
    print("Palette:", ", ".join(LAB_PALETTE))
